config.go

```diff
@@ -331,6 +331,17 @@ func (b *Builder) SetUser(spec string) {
 	b.Docker.Config.User = spec
 }
 
+// OnBuild returns the OnBuild value from the container.
+func (b *Builder) GetOnBuild() []string {
+	return b.Docker.Config.OnBuild
+}
+
+// SetOnBuild sets a trigger instruction to be executed when the image is used
+// as the base of another image.
+func (b *Builder) SetOnBuild(onBuild []string) {
+	b.Docker.Config.OnBuild = onBuild
```

Comment:
Please use `copyStringSlice()` here to make a copy of `onBuild`.  We don't want people to modify the slice after we get it, and assuming that that's a supported way of updating the value we're holding.  (It looks like `Shell()` and `SetShell()` need similar fixes.  It's not a part of this PR, but I don't want to forget to fix that.)